### ISSUE TITLE
Feedback on progress in challenge-type achievements

### ISSUE THREAD

**Issue by timdams** [jonasswiatek/strokes#32]
Provide feedback to the user when he's trying to solve a challenge. For example tell user what test failed or Pex4fun style:
"tried following input; expected this but instead got this"

**Comment by jonasswiatek** [comment]
Good idea. I'll see what I can make when I've merged nrefactory5 into master

**Comment by jonasswiatek** [comment]
The first half of this is addressed now (see the commit message of 8a7b639) - link above.
CalculatorChallenge implements feedback as an example.

It's time to think about how to relay this feedback to the user.
